src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -87,7 +88,7 @@ protected override bool PreExecuteInstruction()
         {
             if (CurrentContext.InstructionPointer >= CurrentContext.Script.Length)
                 return true;
-            return AddGas(OpCodePrices[CurrentContext.CurrentInstruction.OpCode]);
+            return AddGas(NativeContract.Fee.GetOpCodePrice(CurrentContext.CurrentInstruction.OpCode, Snapshot));
```

Comment:
We should use GetRatio only, otherwise we will lose a big performance, other way is caching the prices, and refresh this cache when it's needed.

Reply:
We're concerned that some instructions' cost may be too high, which may need to be adjusted in the future.